app/controllers/users_controller.rb

```diff
@@ -50,6 +56,55 @@ def require_login
   end
   private :require_login
 
+  def check_passive_login
+    if !user_signed_in?
+      redirect_to nyu_shibboleth_omniauth_authorize_path and return if shib_session_exists?
+      if !cookies[:_check_passive_shibboleth]
+        cookies[:_check_passive_shibboleth] = true
+        redirect_to passive_shibboleth_url
+      end
+    end
+  end
+
+  def shib_session_exists?
```

Comment:
@hab278 Throw all helper functions below `check_passive` and `check_passive_shibboleth` and make `private` with an indented

```ruby

  def check_passive
  end

 private

 YOUR PRIVATE FUNCTIONS HERE
```